tests/geometry/test_intersections.py

```diff
@@ -1,9 +1,2539 @@
+                return False
+        return True
+
+    def _arrays_equal(self, a, b):
+        return self._a_in_b(a, b) and self._a_in_b(b, a)
+
+    def test_completely_outside_lower(self):
+        self.setup()
+        f = self.f_1
+        f.pts[0] -= 2
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        assert len(network.fractures) == (0 + 6)
+
+    def test_outside_west_bottom(self):
+        self.setup()
+        f = self.f_1
+        f.pts[0] -= 0.5
+        f.pts[2] -= 1.5
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        assert len(network.fractures) == (0 + 6)
+
+    def test_intersect_one(self):
+        self.setup()
+        f = self.f_1
+        f.pts[0] -= 0.5
+        f.pts[2, :] = [0.2, 0.2, 0.8, 0.8]
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        p_known = np.array(
+            [[0.0, 0.5, 0.5, 0], [0.5, 0.5, 0.5, 0.5], [0.2, 0.2, 0.8, 0.8]]
+        )
+        assert len(network.fractures) == (1 + 6)
+        p_comp = network.fractures[0].pts
+        assert self._arrays_equal(p_known, p_comp)
+
+    def test_intersect_two_same(self):
+        self.setup()
+        f = self.f_1
+        f.pts[0, :] = [-0.5, 1.5, 1.5, -0.5]
+        f.pts[2, :] = [0.2, 0.2, 0.8, 0.8]
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        p_known = np.array([[0.0, 1, 1, 0], [0.5, 0.5, 0.5, 0.5], [0.2, 0.2, 0.8, 0.8]])
+        assert len(network.fractures) == (1 + 6)
+        p_comp = network.fractures[0].pts
+        assert self._arrays_equal(p_known, p_comp)
+
+    def test_incline_in_plane(self):
+        self.setup()
+        f = self.f_1
+        f.pts[0] -= 0.5
+        f.pts[2, :] = [0, -0.5, 0.5, 1]
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        p_known = np.array(
+            [[0.0, 0.5, 0.5, 0], [0.5, 0.5, 0.5, 0.5], [0.0, 0.0, 0.5, 0.75]]
+        )
+        assert len(network.fractures) == (1 + 6)
+        p_comp = network.fractures[0].pts
+        assert self._arrays_equal(p_known, p_comp)
+
+    def test_full_incline(self):
+        self.setup()
+        p = np.array([[-0.5, 0.5, 0.5, -0.5], [0.5, 0.5, 1.5, 1.5], [-0.5, -0.5, 1, 1]])
+        f = pp.PlaneFracture(p, check_convexity=False)
+        network = pp.create_fracture_network([f])
+        network.impose_external_boundary(self.domain)
+        p_known = np.array(
+            [[0.0, 0.5, 0.5, 0], [5.0 / 6, 5.0 / 6, 1, 1], [0.0, 0.0, 0.25, 0.25]]
+        )
+        assert len(network.fractures) == (1 + 6)
+        p_comp = network.fractures[0].pts
+        assert self._arrays_equal(p_known, p_comp)
+
+
+class TestIntersectionPolygonsEmbeddedIn3d:
+    def test_single_fracture(self):
```

Comment:
This is not really a fracture. In this context, aren't we really just testing intersecting polygons?

Reply:
Yeah it would be more accurate saying polygons, I'll change it wherever necessary